presto-native-execution/presto_cpp/main/PeriodicMemoryChecker.h

```diff
@@ -76,6 +76,11 @@ class PeriodicMemoryChecker {
   /// Stops the 'PeriodicMemoryChecker'.
   void stop();
 
+  /// Returns the last known 'current' system memory usage.
+  int64_t currentMemBytes() const {
```

Comment:
uint64_t to align with declared type

Reply:
Why do you need this member and method? systemUsedMemoryBytes() has what's needed?

Reply:
> uint64_t to align with declared type

Original systemUsedMemoryBytes() returns int64_t.
I added atomic uint64_t.
I will sync all three types.

Reply:
> Why do you need this member and method? systemUsedMemoryBytes() has what's needed?
Because systemUsedMemoryBytes() is actually retrieveing the number from the OS and it is not cheap.
This get method can be called by anyone at any time and simply returns the last stored number.

So we will have (already have):
1. Periodic task that fetches the number and stores it in the atomic int.
2. Any other thread can read from this atomic int.

Reply:
I don't see this method used. Is it planned to be used in some followup use case? If not can we add it when it's actually needed?

Reply:
Yes, it will be used in Meta's fbcode - it creates its own checker, which is a member of the class inherited from the PrestoServer.